Library/Homebrew/requirements/java_requirement.rb

```diff
@@ -45,16 +30,83 @@ def inspect
 
   def display_s
     if @version
-      if @version[-1] == "+"
-        op = ">="
-        version = @version[0, @version.length-1]
-      else
+      if exact_version?
         op = "="
-        version = @version
+      else
+        op = ">="
       end
-      "#{name} #{op} #{version}"
+      "#{name} #{op} #{version_num}"
     else
       name
     end
   end
+
+  private
+
+  def version_num
+    if exact_version?
+      @version
+    else
+      @version[0, @version.length-1]
+    end
+  end
+
+  def exact_version?
+    @version && @version[-1] != "+"
+  end
+
+  def java_cmd
+    return false unless File.executable?("/usr/libexec/java_home")
+    args = %w[--failfast]
+    args << "--version" << @version.to_s if @version
+    java_home = Utils.popen_read("/usr/libexec/java_home", *args).chomp
+    return false unless $?.success?
+    Pathname.new(java_home)/"bin/java"
+  end
+
+  def setup_java
+    ohai "setup_java"
+    java = preferred_java
+    return unless java
+    @java = java
+    @java_home = java.dirname.dirname
```

Comment:
There is a very off chance of this blowing up if someone has a very bizarre setup, say like having `/java` as their preferred Java command, but I personally think we don't need to worry about such odd cases.